### DISCUSSION TITLE
Airbyte asset launchpad showing raw values instead of env var names

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17842] [Q&A]
When I load up the launchpad for an airbyte asset, I am still seeing the raw values for actually connecting to the instance (rather than the hidden env values). Anyone have thoughts on why?

It seems to be working fine for all the other resources.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C047VPGQUTV/p1699479709935369?thread_ts=1699479709.935369&cid=C047VPGQUTV).*

**Comment by brentshulman-silkline** [comment]
![image](https://github.com/dagster-io/dagster/assets/134343004/a09e7012-7a3e-4af8-8871-0d567681356b)
![image](https://github.com/dagster-io/dagster/assets/134343004/d07ba69d-5c4a-4cff-bfe6-d05885e8fc97)

**Comment by jamiedemaria** [comment]
Are you doing something like this?

```python
AIRBYTE_HOST = EnvVar("AIRBYTE_HOST")
# same for AIRBYTE_PASSWORD, etc

airbyte_instance = AirbyteResource(host=AIRBYTE_HOST, ...) 

```

If you do this instead, are the values hidden?
```python
airbyte_instance = AirbyteResource(host=EnvVar("AIRBYTE_HOST"), ...) 

```

**Reply by brentshulman-silkline** [reply]
Yes, the former is what I am currently doing. I am not sure why that would make a difference?

**Reply by jamiedemaria** [reply]
@benpankow  knows more about the specifics of when `EnvVar` is evaluated, but it might be pulling the environment variable when the file is interpreted by python instead of when dagster executes the asset

**Reply by brentshulman-silkline** [reply]
Got it. That is kind of what I figured. That docs just say to do it this way :/

**Reply by jamiedemaria** [reply]
the docs say to create the `EnvVars` like in the first code snippet? Can you link the docs page you're looking at so we can fix?

**Reply by brentshulman-silkline** [reply]
Sorry, a bit of confusion on my end. You are correct that class docs say to do it the inline way (`AirbyteResource(host=EnvVar("AIRBYTE_HOST"), ...)`)

However, after changing my code to match this, I am still experiencing this issue.
![image](https://github.com/dagster-io/dagster/assets/134343004/0216ac37-8fe7-49d3-a8bf-266a593c15b1)
![image](https://github.com/dagster-io/dagster/assets/134343004/26b0efa9-2070-4cc1-b875-5cc1fdb235a1)
<img width="792" alt="image" src="https://github.com/dagster-io/dagster/assets/134343004/807e0209-d554-404d-a727-e4c2869e7050">

**Reply by jamiedemaria** [reply]
that's strange. I'm not able to replicate it on my side with a basic setup. What version of dagster are you on? For reference, this is the exact code I'm running. If you want to run it yourself and share what you see, that could help with debugging

```python
class ResourceWithConfig(ConfigurableResource):
    my_secret_value: str

@asset
def my_asset(context, my_resource: ResourceWithConfig):
    context.log.info(my_resource.my_secret_value)

defs = Definitions(
    assets=[my_asset],
    resources={
        "my_resource": ResourceWithConfig(my_secret_value=EnvVar("MY_SECRET_VALUE"))
    }
)
```

**Reply by brentshulman-silkline** [reply]
Well I am not putting it in the resource. I am using `load_assets_from_airbyte_instance` like this:

![image](https://github.com/dagster-io/dagster/assets/134343004/380b28aa-18a6-40e6-b359-e2386441a61d)


Could that have something to do with it?

**Reply by jamiedemaria** [reply]
yeah that could definitely be it. tbh this is getting a bit deeper into the airbyte integration than i'm familiar with, so ben will be the best resource here

**Reply by benpankow** [reply]
Hi Brent, I think this is a quirk with the way that `load_assets_from_airbyte_instance` is initializing the resource. Taking a look, I think this is a bug on our end.

**Reply by brentshulman-silkline** [reply]
Hey @benpankow - any luck?

**Reply by benpankow** [reply]
Hi Brent, https://github.com/dagster-io/dagster/pull/17922 should fix, which will go out with the next release!

**Reply by brentshulman-silkline** [reply]
When is the next release?
